extension/data.ts

```diff
@@ -63,7 +63,7 @@ type DictEntryData = typeof defaultDictEntryData;
 interface Deinflection {
   word: string;
   type: number;
-  reason: string;
+  reason: string | null;
```

Comment:
I've not looked closely yet but I'm always skeptical about adding nullable types. For example, strings can usually be empty with similar meaning as null.

Reply:
Looks like this was the main cause of the issue. Thank you for pointing it out.
Removed the `| null` and flipped the logic to check if the character isKana or not.

Reply:
I've actually added this back as there is a check towards the end that returns `null` that was present in the original legacy code. Let me know if there is another way you'd like to handle this.